util/util.go

```diff
@@ -0,0 +1,56 @@
+package util
+
+import (
+	"bufio"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	separatorRegExp = regexp.MustCompile(`^#+ *(?:-+|=+).*$`)
```

Comment:
It matches with comments like this:

```sh
###
# Header
###
```

So my suggestion is to change the pattern to `^#+\s*[-=]{3,}$`